dali/pipeline/executor/executor2/exec_node_task.cc

```diff
@@ -278,16 +278,13 @@ void OpTask::SetWorkspaceInputs() {
   auto process_input = [&](int i, auto backend) {
     using Backend = decltype(backend);
     const auto &inp = TaskInput<Backend>(ti);
-    // If the output order of the operator is `host` then we don't wait for GPU
-    // inputs - they can't be accessed directly on host and the operator will
-    // have to issue some sort of synchronization if and when necessary.
-    // This optimization is essential to avoid oversynchronization
-    // when the operator needs to access the metadata only (e.g. getting the shape).
```

Comment:
Here we get rid of the dangerous special treatment of GPU inputs to host-ordered operators. The new `metadata` flag takes care of the optimization now.